redbox-core/redbox/models/chain.py

```diff
@@ -261,14 +267,19 @@ def tool_calls_reducer(current: ToolState, update: ToolState | None) -> ToolStat
     return reduced
 
 
-class RedboxState(MessagesState):
-    request: Required[RedboxQuery]
-    documents: Annotated[NotRequired[DocumentState], document_reducer]
-    route_name: NotRequired[str | None]
-    tool_calls: Annotated[NotRequired[ToolState], tool_calls_reducer]
-    metadata: Annotated[NotRequired[RequestMetadata], metadata_reducer]
-    citations: NotRequired[list[Citation] | None]
-    steps_left: Annotated[NotRequired[int], RemainingStepsManager]
+class RedboxState(BaseModel):
+    request: RedboxQuery
+    documents: Annotated[DocumentState, document_reducer] = DocumentState()
+    route_name: str | None = None
+    tool_calls: Annotated[ToolState | None, tool_calls_reducer] = None
+    metadata: Annotated[RequestMetadata | None, metadata_reducer] = None
+    citations: list[Citation] | None = None
+    steps_left: Annotated[int | None, RemainingStepsManager] = None
+    messages: Annotated[list[AnyMessage], add_messages] = Field(default_factory=list)
+
+    @property
+    def last_message(self) -> AnyMessage:
+        return self.messages[-1]
```

Comment:
I think there are probably more nice things we can add like this such as `flat_docs` or `total_token_count`

Reply:
do we need to catch the edge case where there are no messages?